repositories/implementations/csv_mlb_player.go

```diff
@@ -0,0 +1,68 @@
+package repositories
+
+import (
+	"encoding/csv"
+	"errors"
+	"log"
+	"os"
+	"strconv"
+
+	e "github.com/EloYaniel/academy-go-q42021/entities"
+)
+
+// CSVMLBPlayerRepository implements MLBPlayerRepository interface
+type CSVMLBPlayerRepository struct{}
+
+func (repo *CSVMLBPlayerRepository) GetMLBPlayers() ([]e.MLBPlayer, error) {
+	f, err := os.Open("data/mlb_players.csv")
+
+	if err != nil {
+		log.Println("error opening the file", err)
+		return nil, errors.New("error opening the file")
+	}
+	defer f.Close()
+	data, err := csv.NewReader(f).ReadAll()
+
+	if err != nil {
+		log.Println("error reading the file: ", err)
+		return nil, errors.New("error reading the file")
+	}
+	var players []e.MLBPlayer
+	for i, line := range data {
+		if i != 0 {
+			id, _ := strconv.Atoi(line[0])
+			height, _ := strconv.Atoi(line[4])
+			weight, _ := strconv.ParseFloat(line[5], 32)
+			age, _ := strconv.ParseFloat(line[6], 32)
+
+			players = append(players, e.MLBPlayer{
+				ID:       id,
+				Name:     line[1],
+				Team:     line[2],
+				Position: line[3],
+				Height:   height,
+				Weight:   float32(weight),
+				Age:      float32(age),
+			})
+		}
+
+	}
+
+	return players, nil
+}
+
+func (repo *CSVMLBPlayerRepository) GetMLBPlayerByID(id int) (*e.MLBPlayer, error) {
+	players, err := repo.GetMLBPlayers()
+
+	if err != nil {
+		return nil, errors.New("error getting player")
+	}
+
+	for _, p := range players {
+		if p.ID == id {
+			return &p, nil
+		}
+	}
+
+	return &e.MLBPlayer{}, nil
```

Comment:
When a function returns a pointer to a struct and an error, like you have it here, and combined with these particular case where no error was generated and you didn't find the MLBPlayer you should `return nil, nil`.

This way, on the function that calls this one you call check that if your error and player are nil, then no player was found, and then since we identified this case you could return an appropriate error to the user indicating that no player was found